TinCanTests/ActivityTest.cs

```diff
@@ -40,12 +40,16 @@ public void TestActivityIdCase()
         }
 
         [Test]
-        [ExpectedException("System.UriFormatException")]
         public void TestActivityIdInvalidUri()
         {
-            var activity = new Activity();
-            string invalid = "foo";
-            activity.id = invalid;
+            Assert.Throws<System.UriFormatException>(
```

Comment:
I'm :+1: on this but I'm going to wait a bit to see if we can get @mlefoster to weigh in.